platform/linux-generic/odp_packet_io.c

```diff
@@ -2869,7 +2988,27 @@ int odp_pktout_send(odp_pktout_queue_t queue, const odp_packet_t packets[],
 	if (_ODP_PCAPNG)
 		_odp_dump_pcapng_pkts(entry, queue.index, packets, num);
 
-	return entry->s.ops->send(entry, queue.index, packets, num);
+	if (odp_unlikely(_odp_pktio_tx_compl_enabled(entry))) {
+		for (int i = 0; i < num; i++)
+			check_tx_compl_ev(packet_hdr(packets[i]), i, tx_compl_info, &num_tx_cevs);
+
+		if (num_tx_cevs > 0) {
+			const int num_alloc = odp_buffer_alloc_multi(entry->s.tx_compl_pool, bufs,
+								     num_tx_cevs);
```

Comment:
By the spec alloc_multi may return <0 (on failure). Seems that implementation does not do that, but crashes if pool is invalid.

Maybe a comment could be added saying that num_alloc >= 0 is assumed. I think LTO and static checkers would warn if we change alloc implementation later to return -1 (on invalid pool). You could try that with a LTO build.

Reply:
Ended up handling < 0 case in v6.